projects/scion/microfrontend-platform/src/lib/microfrontend-platform.ts

```diff
@@ -143,6 +147,7 @@ export class MicrofrontendPlatform {
         registerRunlevel2Initializers();
         registerRunlevel3Initializers();
 
+        Beans.registerIfAbsent(VERSION, {useValue: version, destroyOrder: Number.MAX_VALUE});
```

Comment:
Correct, we use `registerIfAbsent ` to override the version in tests.

I like your idea of setting the version in an initializer. But, a client connects in runlevel 0 to the host, thus, the version needs to be set in that runlevel. Since initializers are executed in registration order, we can register an initializer to be executed in runlevel 0 prior to starting the platform. That way, we can override the default version.